### ISSUE TITLE
yarn-client: undefined symbol: _Py_write_noraise

### ISSUE THREAD

**Issue by bestfleer** [intel-analytics/BigDL#5078]
I use spark-submit-with-dllib to submit my task to yarn, but got an jep error:
creating: createTorchLoss
creating: createTorchOptim
creating: createAUC
creating: createZooKerasAccuracy
creating: createEstimator
creating: createMaxEpoch
22-07-12 11:35:57 [Thread-9] INFO  InternalDistriOptimizer$:987 - TorchModel[a02e566c] isTorch is true
22-07-12 11:35:57 [Thread-9] INFO  InternalDistriOptimizer$:993 - torch model will use 1 OMP threads.
22-07-12 11:35:57 [Thread-9] INFO  DistriOptimizer$:826 - caching training rdd ...
22-07-12 11:36:05 [Thread-9] INFO  DistriOptimizer$:652 - Cache thread models...
22-07-12 11:36:06 [Thread-9] INFO  DistriOptimizer$:654 - Cache thread models... done
22-07-12 11:36:06 [Thread-9] INFO  DistriOptimizer$:164 - Count dataset
22-07-12 11:36:06 [Thread-9] INFO  DistriOptimizer$:168 - Count dataset complete. Time elapsed: 0.094004416s
22-07-12 11:36:06 [Thread-9] INFO  DistriOptimizer$:176 - config  {
        computeThresholdbatchSize: 100
        maxDropPercentage: 0.0
        warmupIterationNum: 200
        isLayerwiseScaled: false
        dropPercentage: 0.0
 }
22-07-12 11:36:06 [Thread-9] INFO  DistriOptimizer$:180 - Shuffle data
22-07-12 11:36:06 [Thread-9] INFO  DistriOptimizer$:183 - Shuffle data complete. Takes 0.024611782s
2022-07-12 11:36:07 WARN  TaskSetManager:69 - Lost task 0.0 in stage 93.0 (TID 4169) (gdios-hadoop02 executor 2): jep.JepException: jep.JepException: <class 'ImportError'>: /app/yarn/nm/usercache/root/appcache/application_1657596173116_0002/container_e755_1657596173116_0002_01_000003/enviroment/lib/python3.7/lib-dynload/_posixsubprocess.cpython-37m-x86_64-linux-gnu.so: undefined symbol: _Py_write_noraise

The submit shell is:
export PYSPARK_DRIVER_PYTHON=/opt/anaconda3/envs/bigdl/bin/python
PYSPARK_PYTHON=enviroment/bin/python spark-submit-with-dllib \
--master yarn \
--deploy-mode client \
--executor-memory 4g \
--driver-memory 4g \
--executor-cores 2 \
--num-executors 2 \
--archives hdfs:///tmp/enviroment.tar.gz#enviroment \
--conf spark.yarn.appMasterEnv.PYSPARK_PYTHON=enviroment/bin/python \
--conf spark.executorEnv.PYSPARK_PYTHON=enviroment/bin.python \
--conf spark.executorEnv.PYTHONHOME=enviroment \
--conf spark.executorEnv.PYTHONPATH=enviroment/bin \
examples/bigdl_demo.py \

can you help me to solve this problem? thanks

**Comment by qiuxin2012** [comment]
Pytorch on yarn with dllib backend should use `init_orca_on_yarn` to init sparkcontext, and use `python examples/bigdl_demo.py` directly. Spark-submit is not supported right now.
See https://bigdl.readthedocs.io/en/latest/doc/Orca/QuickStart/orca-pytorch-quickstart.html for a quickstart.

**Comment by bestfleer** [comment]
> Pytorch on yarn with dllib backend should use `init_orca_on_yarn` to init sparkcontext, `cluster_mode` can be `yarn-client` or `yarn-cluster`, and use `python examples/bigdl_demo.py` directly. Spark-submit is not supported right now.
> See https://bigdl.readthedocs.io/en/latest/doc/Orca/QuickStart/orca-pytorch-quickstart.html for a quickstart.

Thanks for your response. My code can work on cluster_mode=yarn-cilent now, but I met a new problem when I set cluster_mode=yarn-cluster:
2022-07-14 09:25:27 WARN  ScriptBasedMapping:254 - Exception running /etc/hadoop/conf/custom_topology_script.py 192.168.12.196 192.168.12.195 
ExitCodeException exitCode=1: Fatal Python error: _PyMainInterpreterConfig_Read: memory allocation failed
Can you help me figure out? Thanks for your help

**Comment by qiuxin2012** [comment]
I tried the `yarn-cluster` model, also get the same warning, the error message is just a warning, your job should run correctly. You can print some success log at the end of your code.
I will try find a way to disable the warning. The warning is occured when fetch the rack topology.

**Comment by qiuxin2012** [comment]
@bestfleer 
The warning is caused by  `/etc/hadoop/conf/custom_topology_script.py`, this script throws exception as we has changed some pyspark python setting.

I find a workaround to disable this warning, you can change the `net.topology.node.switch.mapping.impl` to `org.apache.hadoop.net.TableMapping`, and set new property `net.topology.table.file.name` in you hadoop core-site.xml, like:
```
    <property>
        <name>net.topology.node.switch.mapping.impl</name>
        <value>org.apache.hadoop.net.TableMapping</value>
    </property>
    <property>
        <name>net.topology.table.file.name</name>
	<value>/opt/hadoop/latest/etc/hadoop/topology.tbl</value> # change to your path
    </property>
```
`topology.tbl` should be a ip-rack mapping like:
```
host1  /default-rack
host2  /default-rack
host3  /other-rack
```
